src/spritecache.cpp

```diff
@@ -669,203 +649,125 @@ bool LoadNextSprite(int load_index, SpriteFile &file, uint file_sprite_id)
 	return true;
 }
 
-
 void DupSprite(SpriteID old_spr, SpriteID new_spr)
 {
 	SpriteCache *scnew = AllocateSpriteCache(new_spr); // may reallocate: so put it first
 	SpriteCache *scold = GetSpriteCache(old_spr);
 
 	scnew->file = scold->file;
 	scnew->file_pos = scold->file_pos;
-	scnew->ptr = nullptr;
+	scnew->ClearSpriteData();
 	scnew->id = scold->id;
 	scnew->type = scold->type;
 	scnew->warned = false;
+	scnew->control_flags = scold->control_flags;
 }
 
-/**
- * S_FREE_MASK is used to mask-out lower bits of MemBlock::size
- * If they are non-zero, the block is free.
- * S_FREE_MASK has to ensure MemBlock is correctly aligned -
- * it means 8B (S_FREE_MASK == 7) on 64bit systems!
- */
-static const size_t S_FREE_MASK = sizeof(size_t) - 1;
-
-/* to make sure nobody adds things to MemBlock without checking S_FREE_MASK first */
-static_assert(sizeof(MemBlock) == sizeof(size_t));
-/* make sure it's a power of two */
-static_assert((sizeof(size_t) & (sizeof(size_t) - 1)) == 0);
-
-static inline MemBlock *NextBlock(MemBlock *block)
-{
-	return (MemBlock*)((byte*)block + (block->size & ~S_FREE_MASK));
-}
-
-static size_t GetSpriteCacheUsage()
+static void DeleteEntriesFromSpriteCache(size_t target)
```

Comment:
nitpick: maybe add a doxygen comment here what `target` does. One can read it as: the size, the count, the amount to remove, the size to remove, etc. It is a bit ambiguous. But nothing a comment cannot fix :)